### PR TITLE
feat: add plugin zentao

### PR DESCRIPTION
**Pull request by hxcGit**
## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [ ] I have read through the [CONTRIBUTING.md](../CONTRIBUTING.md) documentation.
- [ ] My code has the necessary comments and documentation (if needed).
- [ ] I have added relevant tests

### Add zentao plugin, currently just deal with zentao installation.
For this implementation zentao application is deployed in a k8s cluster by client-go, later I expect to deploy zentao by Helm.

##### 1. create.go
Create zentao namespaces, persistentVolume, persistentVolumeClaim, deployment, service.
- PersistentVolume use `HostPath` and in a single-node cluster now.
- Service is exposed by `NodePort`.

##### 2. delete.go
Delete zentao namespaces, persistentVolume, persistentVolumeClaim, deployment, service.
- Perform the opposite operation of creation.

##### 3. k8s.go
Some cluster-related operations

##### 4. update.go
Just rebuild zentao deployment and service.

### Options and configuration
Here needs more advice and consideration. I want to support different installation way in config files and use `type` field to indicate.
A simple example configuration file is as follows:
```yaml
tools:
  # name of the tool
  - name: zentao
    # id of the tool instance
    instanceID: default
    # format: name.instanceID; If specified, dtm will make sure the dependency is applied first before handling this tool
    dependsOn: [ ]
    # options for the plugin
    options:
      type: "clientAPI"
      clientAPI:
        namespace: "zentao"
        storageClassName: "zentao-storage"
        persistentVolume:
          zentaoPVName: "zentao-pv"
          zentaoPVCapacity: "1G"
          mysqlPVName: "mysql-pv"
          mysqlPVCapacity: "1G"
        persistentVolumeclaim:
          zentaoPVCName: "zentao-pvc"
          zentaoPVCCapacity: "1G"
          mysqlPVCName: "mysql-pv"
          mysqlPVCCapacity: "1G"
        deployment:
          name: "zentao-dp"
          replicas: 3
          image: "easysoft/zentao:latest"
          mysqlPasswdName: "MYSQL_ROOT_PASSWORD"
          mysqlPasswdValue: "1234567"
        service:
          name: "zentao-svc"
          nodePort: 30081
      helm:
```

By the way, I am not particularly satisfied with the current implementation, I need more requirements and suggestions.

## Related Issues

#508 

## New Behavior (screenshots if needed)

image

### CODE REVIEW

**Comment by KeHaohaoke** [comment]
I used your code for a local test, and it can run well. It's good that you try to specify the deployment method through configuration. Documents can be completed. @hxcGit

**Comment by hxcGit** [comment]
/hold 
I'll check the code again and add plugin docs.

File: internal/pkg/plugin/zentao/zentao.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package zentao
```
Comment (IronCore864) [inline internal/pkg/plugin/zentao/zentao.go:1]: No newline at end of file. Please do a 'go fmt'.

**Comment by IronCore864** [comment]
It seems MySQL is a requirement for zentao. Do you think it's possible to install MySQL as well? @hxcGit @daniel-hutao @KeHaohaoke

**Comment by hxcGit** [comment]
@IronCore864 Yep, Zentao need MySQL.
1. For official images `easysoft/zentao`, zentao service and mysql service are all installed. https://www.zentao.net/book/zentaopmshelp/405.html
2. I'm not exactly sure what you mean. If we use a individual mysql service in another pod, we will need to modify some initialization scripts of the official image accordingly. 

@KeHaohaoke @daniel-hutao  any good ideas?

File: internal/pkg/plugin/zentao/zentao.go (hunk 1 above)
Comment (hxcGit) [reply internal/pkg/plugin/zentao/zentao.go:1]: I'll fix it.

**Comment by hxcGit** [comment]
By the way, the official image doesn't seem to work well in `mac m1 arm`. Please test in linux amd64 system.

**Comment by hxcGit** [comment]
How to reopen this pull request?

**Comment by hxcGit** [comment]
@KeHaohaoke @IronCore864  PTAL.

**Comment by KeHaohaoke** [comment]
Zentao's official image integrates MySQL, so we discuss that we can use the official image first. That's ok. @hxcGit

File: internal/pkg/show/config/plugins/zentao.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        # initial password name for mysql database, you can specify any name you like
+        mysqlPasswdName: 'MYSQL_ROOT_PASSWORD'
+        # initial password value for mysql database, you can specify any value you like
+        mysqlPasswdValue: '1234567'
+      # zentao application is exposed via K8S Service
+      service:
+        # name of zentao service
+        name: 'zentao-svc'
+        # nodePort of zentao service, currently zentao plugin only support `nodePort` type
+        nodePort: 30081
```
Comment (KeHaohaoke) [inline internal/pkg/show/config/plugins/zentao.yaml:51]: No newline at the end of the file zentao.yaml. Please check it.

Comment (KeHaohaoke) [reply internal/pkg/show/config/plugins/zentao.yaml:51]: @hxcGit 

> No newline at the end of the file zentao.yaml. Please check it.

Comment (hxcGit) [reply internal/pkg/show/config/plugins/zentao.yaml]: will fix it

**Comment by IronCore864** [comment]
@hxcGit thanks for this great contribution. I'll test it locally and get back to you ASAP.

**Comment by IronCore864** [comment]
Merging now, see discussions in Slack. Please contribute another PR to follow up on those discussions. Thanks for your contribution! @hxcGit

**Review by IronCore864 — APPROVED** [review]

**Pull request merged**